tscached/shadow.py

```diff
@@ -0,0 +1,33 @@
+import logging
+
+
+def should_add_to_readahead(config, referrer, headers):
+    """ Should we add this KQuery for readahead behavior?
+        config: dict representing the top-level tscached config
+        referrer: str, from the http request
+        headers: dict, all headers from the http request
+        returns: boolean
+    """
+    if headers.get(config['shadow']['http_header_name'], None):
+        return True
+
+    for substr in config['shadow']['referrer_blacklist']:
+        if substr in referrer:
+            return False
+    return True
```

Comment:
Yea that makes sense. I was just concerned about maintaining read ahead for every dashboard that was ever viewed, even if it's no longer being actively viewed.